### DISCUSSION TITLE
feat: create portableapps version

### DISCUSSION THREAD

**Discussion by mr-cheffy** [zen-browser/desktop#804] [Ideas]
### What feature would you like to see?

I am a current user of regular Firefox, and I also use Firefox from PortableApps on a USB drive to use on college computers for personal security and convenience, as I use a few extensions that are personally important.

Mozilla has given PortableApps explicit permission to package the browser under the `Mozilla Firefox` brand as a portable version, however I believe that in Zen Browser's current state, this permission would not be needed. Thus, I would appreciate it if a PortableApps version were to be packaged.

### What platform are you seeing the problem on?

Windows

### Additional context

https://portableapps.com/

https://portableapps.com/apps/internet/firefox_portable
 
 ~ _Created automatically from issue: https://github.com/zen-browser/desktop/issues/536_

**Comment by Sirmadsen** [comment]
I second this.

**Comment by ioqbit** [comment]
I like PortableApp too.

I started using it in 2018, which is 7 years ago. Initially, I only needed to bring a USB flash drive to use my data on different computers at school. Later, I also installed it on my home computer because the software did not need to be re-downloaded and reconfigured after reinstalling the system.

I think most software should follow the portable design, and the data only stays in the software's own directory, which is convenient for users to use in different environments.

For some small-function software, I use https://github.com/Bioruebe/UniExtract2 to make a portable version (but I cannot control the software itself to read and write data in the system directory)

![image](https://github.com/user-attachments/assets/953ee6c9-8995-44cd-ac79-c57fc0109426)

**Comment by LifeCANvs** [comment]
I have stopped using software that require installation, so unfortunately this prevents me from using Zen browser on Windows :(
It is actually surprising that it is only available as an installer package for Windows, while multiple portable options are available for Linux systems (on which I happily use it with the tarball version, and an AppImage is also available). So it seems that I need to continue using Firefox on my Windows system due to the lack of a portable Zen option for this platform. I hope this issue will be addressed soon, to give Zen users more freedom of choice, and flexibility with how and where to use it on and across our systems.

**Comment by Jobima1st** [comment]
is it that hard to do official portable? what about taking code/ideas from official portable librewolf or floorp (I think it use libportable)

**Comment by LifeCANvs** [comment]
@zen-browser devs : Could your team use these MIT licensed scripts to create an **official** portable Windows version of the current Zen browser?
https://github.com/wysh3/Zen-Browser-Portable
Yes, I will use the scripts myself to build my own portable version of Zen for use on Windows, but it requires quite a few steps on a Linux system (the scripts have not been created for use on Windows systems), so most people who want to run Zen on Windows will probably not (be able to) do this. So an official ready-built portable Zen download for Windows would certainly be much preferred.

**Comment by RaXorX** [comment]
Welcome to a calmer internet - with a tinge of drama. Heh. Recently wanted to switch from chromium to firefox, been planning to for months but never got around it until now. Had Zen, Floorp and LibreW as my alternatives to test out. Came across the Zen with no portable builds after already downloading Floorp and LibreWolf as portables. Went through the issues threads and boy. What a clusterfuck. I had no idea that people had almost zero sense in what the notion of portable or portability means especially in this day and age where portable apps exist. It's like someone had no idea what portable meant in today's time and age and had such lengthy arguments on the same.

Portable > Binary + User data. (optionally including registry data)
Portable Binary > Binary only (Apparently LibreWolf decided to call that a package or as I'd call it, an archived package since it's an archive)

Which is barely any different from extracting the core application files from the executable and running it from a non system or a user folder. Or letting it install and running it from elsewhere. I occassionally do that with eletron apps a lot instead of installing them. How hard is it exactly to make that distinction?

Honestly have no idea why this thread even existed if @mr-cheffy your end conclusion and solution to this was going to be a literal removal of the portable branch itself entirely. I'm honestly curious what exactly went there? Did something happen? On Discord maybe? Sounds like a move out of frustration rather than reasonable thinking.

The dramatic exit of the portable builds genuinely sounds so much like a poor product thinking. Especially when other browsers like Floorp and LibreWolf already tackle that with... no such volatility. They never reacted to user requests in such a way and simply acted, quite positively. 

It's almost like the team behind Zen went through some very drastic development process, where either you devs failed to understand the practicality of what users were requesting. Not understanding it entirely at all. Understanding but not giving a... crap and instead gave it a full on boot. Almost sounds like some sort of ego/philosophy driven decision making - I don't understand it, I don't give a crap, I can't do it well or perfectly so I'll just remove it entirely. 

My question. What exactly happened? Why the dismissal? And whatever pholosophy it is that you actively don't want to actually provide it? Are there any technical limitations or issues you are running into or you think users may run into doing that?

Just giving it a boot really sounds very unreasonable and unprofessional from a highly - esteemed, opiniated browser.

**Reply by mr-cheffy** [reply]
@RaXorX what are you talking about? 😅

**Reply by RaXorX** [reply]
https://github.com/zen-browser/desktop/issues/7270
https://github.com/zen-browser/desktop/issues/563
https://github.com/zen-browser/desktop/issues/344

I think these issues sum the entire thing up pretty much. On April 2025 the portable version was entire closed, with no remark whatsoever about the reasons for it. The only plausible assumption is that the dev team was entirely fed up by the continuous streams of discussion over portable builds not being portable. Couldn't have been any clearer than that. From there, my question stands. @mr-cheffy

**Reply by mr-cheffy** [reply]
I think you are confusing what "portable" means in this case, the current zen builds are the portable ones. Check this out https://github.com/zen-browser/desktop/wiki/Why-have-optimized-builds-been-removed%3F

As a matter of fact, this thread got created BEFORE it got "removed", according to you? Lol

**Reply by RaXorX** [reply]
Why the heck were they called that then my mate? That's the big question. You can educate all you want, but that kind of education is called grooming in layman's terms. Or indoctrination. In a more respectable tone.

Well... In that case I'll reply back with the exact same comment someone else on reddit threw in my face. But hey, it's not me to shame anyone whatever they use for naming convention or anything in particular. And if you think it was more understandable or easier to call those builds as Portable on the website (the issues already have screenshots of the website clearly pointing out the Portable builds. And not just that, instead of pointing this out, in those issues they were simply closed by none other than you.)

**Reply by mr-cheffy** [reply]
Not sure, it was a long time ago

**Reply by RaXorX** [reply]
Yes. Also that you closed all those user request issues for some reason instead of actually providing a portable build, contrary to other browsers providing it. With a lack of clear reason as to why you couldn't or didn't want to. That was all.

**Reply by mr-cheffy** [reply]
Yeah? Thats how bug trackers work

**Reply by RaXorX** [reply]
I thought those were feature requests.. Or well. Not particularly. Cause they called you out on the naming convention and you had nothing else to give. So what about those entire discussion? It exists since what... Aug 2024. I'd assume you have nothing more to say and don't actually want to answer the core of the questions, despite all the unintentionally intentional misunderstandings...  From your naming conventions. The lack of actual portable builds. And the crux is that you don't want to do that. Because that's all I am getting from all this.

**Reply by mr-cheffy** [reply]
Sure

**Reply by RaXorX** [reply]
Cute. :)
